src/coreclr/jit/importer.cpp

```diff
@@ -4782,6 +4788,390 @@ GenTree* Compiler::impIntrinsic(GenTree*                newobjThis,
+            // cpblk
+            // ret
+
+            return nullptr;
+        }
+
+        case NI_SRCS_UNSAFE_CopyBlockUnaligned:
+        {
+            assert(sig->sigInst.methInstCount == 0);
+
+            // ldarg.0
+            // ldarg.1
+            // ldarg.2
+            // unaligned. 0x1
+            // cpblk
+            // ret
+
+            return nullptr;
+        }
+
+        case NI_SRCS_UNSAFE_InitBlock:
+        {
+            assert(sig->sigInst.methInstCount == 0);
+
+            // ldarg.0
+            // ldarg.1
+            // ldarg.2
+            // initblk
+            // ret
+
+            return nullptr;
+        }
+
+        case NI_SRCS_UNSAFE_InitBlockUnaligned:
+        {
+            assert(sig->sigInst.methInstCount == 0);
+
+            // ldarg.0
+            // ldarg.1
+            // ldarg.2
+            // unaligned. 0x1
+            // initblk
+            // ret
+
+            return nullptr;
+        }
+
+        case NI_SRCS_UNSAFE_IsAddressGreaterThan:
+        {
+            assert(sig->sigInst.methInstCount == 1);
+
+            // ldarg.0
+            // ldarg.1
+            // cgt.un
+            // ret
+
+            GenTree* op2 = impPopStack().val;
+            GenTree* op1 = impPopStack().val;
+
+            GenTree* tmp = gtNewOperNode(GT_GT, TYP_INT, op1, op2);
+            tmp->gtFlags |= GTF_RELOP_NAN_UN | GTF_UNSIGNED;
+            return gtFoldExpr(tmp);
+        }
+
+        case NI_SRCS_UNSAFE_IsAddressLessThan:
+        {
+            assert(sig->sigInst.methInstCount == 1);
+
+            // ldarg.0
+            // ldarg.1
+            // clt.un
+            // ret
+
+            GenTree* op2 = impPopStack().val;
+            GenTree* op1 = impPopStack().val;
+
+            GenTree* tmp = gtNewOperNode(GT_LT, TYP_INT, op1, op2);
+            tmp->gtFlags |= GTF_RELOP_NAN_UN | GTF_UNSIGNED;
+            return gtFoldExpr(tmp);
+        }
+
+        case NI_SRCS_UNSAFE_IsNullRef:
+        {
+            assert(sig->sigInst.methInstCount == 1);
+
+            // ldarg.0
+            // ldc.i4.0
+            // conv.u
+            // ceq
+            // ret
+
+#if (REGSIZE_BYTES == 8)
+            bool uns = true;
+#else
+            bool uns = false;
+#endif
+
+            GenTree* op1 = impPopStack().val;
+
+            GenTree* cns = gtNewIconNode(0, TYP_INT);
+            impBashVarAddrsToI(cns);
+            cns = gtNewCastNode(genActualType(TYP_U_IMPL), cns, uns, TYP_U_IMPL);
+
+            GenTree* tmp = gtNewOperNode(GT_EQ, TYP_INT, op1, cns);
+            return gtFoldExpr(tmp);
+        }
+
+        case NI_SRCS_UNSAFE_NullRef:
+        {
+            assert(sig->sigInst.methInstCount == 1);
+
+            // ldc.i4.0
+            // conv.u
+            // ret
+
+#if (REGSIZE_BYTES == 8)
+            bool uns = true;
+#else
+            bool uns = false;
+#endif
+
+            GenTree* cns = gtNewIconNode(0, TYP_INT);
+            impBashVarAddrsToI(cns);
```

Comment:
```suggestion
```

Reply:
Removed this from all the places where it was being called on a `gtNewIconNode` since those will always be a nop.